improver/blending/utilities.py

```diff
@@ -62,3 +78,153 @@ def find_blend_dim_coord(cube, blend_coord):
+    dimension.  Unless these are time-related coordinates, they should be
+    removed after blending.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be blended
+        blend_coord (str):
+            Name of coordinate over which the blend will be performed
+
+    Returns:
+        list:
+            List of names of coordinates to remove
+    """
+    try:
+        (blend_dim,) = cube.coord_dims(blend_coord)
+    except ValueError:
+        # occurs if the blend coordinate is scalar
+        if blend_coord == MODEL_BLEND_COORD:
+            return [copy(MODEL_BLEND_COORD), copy(MODEL_NAME_COORD)]
+        return None
+
+    crds_to_remove = []
+    for coord in cube.coords():
+        if coord.name() in TIME_COORDS:
+            continue
+        if blend_dim in cube.coord_dims(coord):
+            crds_to_remove.append(coord.name())
+    return crds_to_remove
+
+
+def update_blended_metadata(
+    cube,
+    blend_coord,
+    coords_to_remove=None,
+    cycletime=None,
+    attributes_dict=None,
+    model_id_attr=None,
+):
+    """
+    Update metadata as required after blending
+    - For cycle and model blending, set a single forecast reference time
+    and period using current cycletime
+    - For model blending, add attribute detailing the contributing models
+    - Remove scalar coordinates that were previously associated with the
+    blend dimension
+    - Update attributes as specified via process arguments
+    - Set any missing mandatory arguments to their default values
+
+    Modifies cube in place.
+
+    Args:
+        cube (iris.cube.Cube):
+            Blended cube
+        blend_coord (str):
+            Name of coordinate over which blending has been performed
+        coords_to_remove (list of str):
+            Name of scalar coordinates to be removed from the blended cube
+        cycletime (str):
+            Current cycletime in YYYYMMDDTHHmmZ format
+        model_id_attr (str):
+            Name of attribute for use in model blending, to record the names of
+            contributing models on the blended output
+        attributes_dict (dict):
+            Optional user-defined attributes to add to the cube
+    """
+    if blend_coord in ["forecast_reference_time", MODEL_BLEND_COORD]:
+        _set_blended_time_coords(cube, cycletime)
+
+    if blend_coord == MODEL_BLEND_COORD:
+        (contributing_models,) = cube.coord(MODEL_NAME_COORD).points
+        # iris concatenates string coordinates as a "|"-separated string
+        blended_attribute_value = " ".join(sorted(contributing_models.split("|")))
+        cube.attributes[model_id_attr] = " ".join(
+            sorted(contributing_models.split("|"))
+        )
+
+    if coords_to_remove is not None:
+        for coord in coords_to_remove:
+            cube.remove_coord(coord)
+
+    if attributes_dict is not None:
+        amend_attributes(cube, attributes_dict)
+
+    for attr in MANDATORY_ATTRIBUTES:
```

Comment:
You could do a one-liner here:

```suggestion
    cube.attributes.update({key : MANDATORY_ATTRIBUTES[key] for key in filter(lambda attr: attr not in cube.attributes, MANDATORY_ATTRIBUTES)})
```
it's personal preference, so feel free to disregard.